custom_components/rainforest_emu_2/emu2.py

```diff
@@ -101,21 +103,43 @@ async def serial_read(self):
                 break
             
             line = line.decode("utf-8").strip()
+            
+            if "Port already in use" in line:
+                _LOGGER.error("Another client is already connected to the TCP server, disconnect that client and try again.")
+            if len(line) == 0:
+                self._connected = False
+                _LOGGER.error("No Data Received, Disconnecting")
+                await asyncio.sleep(10)
+                await self.open()
+                response = ""
+                               
             _LOGGER.debug("received %d: %s", len(line), line)
-
+              
             response += line
             if line.startswith('</'):
                 try:
-                    self._connected = True
                     self._process_reply(response)
+                    #Upon first recieved message, get the basic data
```

Comment:
This new block is a bit ugly. Is the point here to catch when the socket is disconnected?